message/router.go

```diff
@@ -131,11 +145,18 @@ func (r *Router) Logger() watermill.LoggerAdapter {
 
 // AddMiddleware adds a new middleware to the router.
 //
-// The order of middlewares matters. Middleware added at the beginning is executed first.
+// The order of middlewares matters. middleware added at the beginning is executed first.
 func (r *Router) AddMiddleware(m ...HandlerMiddleware) {
 	r.logger.Debug("Adding middlewares", watermill.LogFields{"count": fmt.Sprintf("%d", len(m))})
 
-	r.middlewares = append(r.middlewares, m...)
+	r.addMiddleware("", true, m...)
+}
+
+func (r *Router) addMiddleware(handlerName string, isRouterLevel bool, m ...HandlerMiddleware) {
```

Comment:
in general IMO it's a good idea to avoid any bool in arguments and create separated functions here (and as I understand we need to pass handlerName only if it's handler level middleware, so it may also simplify it a bit

but in general, I like the design to keep everything in one slice :)